src/pages/view-only.tsx

```diff
@@ -0,0 +1,91 @@
+import { Box, ChakraProvider } from '@chakra-ui/react';
+import { useInterpret } from '@xstate/react';
+import Head from 'next/head';
+import { useRouter } from 'next/router';
+import React, { useEffect, useMemo } from 'react';
+import { createMachine } from 'xstate';
+import { CanvasProvider } from '../CanvasContext';
+import { CanvasView } from '../CanvasView';
+import { EmbedProvider } from '../embedContext';
+import { SimulationProvider } from '../SimulationContext';
+import { simulationMachine } from '../simulationMachine';
+import { theme } from '../theme';
+import { NextComponentWithMeta } from '../types';
+import { useInterpretCanvas } from '../useInterpretCanvas';
+import { parseEmbedQuery, withoutEmbedQueryParams } from '../utils';
+
+const machine = createMachine({
+  initial: 'wow',
+  states: {
+    wow: {
+      on: {
+        NEXT: {
+          target: 'new',
+        },
+      },
+    },
+    new: {},
+  },
+});
+
+const ViewOnlyPage: NextComponentWithMeta = () => {
+  const canvasService = useInterpretCanvas({
+    sourceID: null,
+  });
+  const simulationService = useInterpret(simulationMachine);
+  const router = useRouter();
+
+  useEffect(() => {
+    simulationService.send({
+      type: 'MACHINES.REGISTER',
+      machines: [machine],
+    });
+  }, []);
+
+  const embed = useMemo(
+    () => ({
+      ...parseEmbedQuery(router.query),
+      isEmbedded: true,
+      originalUrl: withoutEmbedQueryParams(router.query),
+    }),
+    [router.query],
+  );
+  return (
+    <>
+      <Head>
+        <script src="https://unpkg.com/elkjs@0.7.1/lib/elk.bundled.js"></script>
+      </Head>
```

Comment:
won't this page need any OG tags? would it be hard to reuse the existing `AppHead` and just configure it to skip downloading Monaco and whatever that is not needed?

Reply:
@Andarist Not sure it will need OG tags as it'll mostly be used for:

1. OG image generation
2. Embedding in PR comments

Maybe it might need it for the second one? Unsure.

Reply:
the second one might especially need them as whenever we share a URL somewhere publicly it can be copied and shared again